### PR TITLE
feat: add unit test for createLBStatus

### PR DESCRIPTION
**Pull request by sakshi-1505**
**What this PR does / why we need it**:

**Which issue this PR fixes(if applicable)**:
refers #2400

**Special notes for reviewers**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by dulek** [comment]
Looks good I think, just needs a rebase.

**Comment by sakshi-1505** [comment]
@dulek please take a look now

**Comment by sakshi-1505** [comment]
bump @dulek @pierreprinetti

**Comment by dulek** [comment]
Please squash the commits before this can be merged.

**Comment by dulek** [comment]
/approve

I'm leaving approval, will lgtm once it's squashed.

**Comment by sakshi-1505** [comment]
@pierreprinetti @dulek Can you please approve this now? I did the rebase, there is addition of another test in this PR because the authors are adding test in b/w of the file rather than appending things at the end which is the major cause of all this conflicts & errors.

File: pkg/openstack/loadbalancer_test.go
Comment (pierreprinetti) [inline pkg/openstack/loadbalancer_test.go:586]: little typo:

```suggestion
			name: "it should return fakehostname if proxyProtocol & IngressHostName is enabled without svc annotation",
```

File: pkg/openstack/loadbalancer_test.go
Comment (pierreprinetti) [inline pkg/openstack/loadbalancer_test.go:565]: Is this setup required for the test case? Why did you choose to include it?

**Review by pierreprinetti — CHANGES_REQUESTED** [review]
Your test is correct, as in: it validates the logic that is currently in place.

However, I think you could walk an extra mile and make it more robust.

Essentially, your test validates two fields of `Status`: `Hostname` and `IP`. Now, imagine that tomorrow we need to add another field: for example (and this is totally made up), `maximumConcurrentConnections`, that defaults to 1000.

As soon as we add that new field, your test fails, because you are comparing for equality an entire `corev1.LoadBalancerStatus` object.

---

One way to make your test future-proof is to only check the properties you are actually testing for. Instead of comparing the entire object, you could check that the single `Hostname` and `IP` properties match what you need them to. Do you want to try that?

File: pkg/openstack/loadbalancer_test.go
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go:565]: Actually, this is included to make sure that if at some later time we change the way of getHostFromSVC annotation the function doesn't go into this code block https://github.com/kubernetes/cloud-provider-openstack/blob/60ad70c0b53d545f323df71b4603bfc8a98dd2df/pkg/openstack/loadbalancer.go#L1881C1-L1882C1. Moreover this is to ensure the declarative behaviour of the unit test as they should be.

File: pkg/openstack/loadbalancer_test.go
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: Done

File: pkg/openstack/loadbalancer_test.go
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: Fixed

File: pkg/openstack/loadbalancer_test.go
Comment (pierreprinetti) [inline pkg/openstack/loadbalancer_test.go:642]: I'd call the function under test just once, save the result in a variable, and assert the variable. I know it's not a big deal in this case, but calling it once per assertion  can be a problematic pattern when the function under test is more expensive.

File: pkg/openstack/loadbalancer_test.go
Comment (pierreprinetti) [inline pkg/openstack/loadbalancer_test.go:641]: What happens if you remove this conditional?

File: pkg/openstack/loadbalancer_test.go
Comment (pierreprinetti) [inline pkg/openstack/loadbalancer_test.go:644]: What happens if you remove this conditional?

File: pkg/openstack/loadbalancer_test.go
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: I don't know why I believed that the want.result would be a pointer where it's a string type itself. I removed both conditional

File: pkg/openstack/loadbalancer_test.go
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: https://github.com/kubernetes/cloud-provider-openstack/pull/2408#discussion_r1361051543

File: pkg/openstack/loadbalancer_test.go
Comment (sakshi-1505) [reply pkg/openstack/loadbalancer_test.go]: done

File: pkg/openstack/loadbalancer_test.go
Comment (pierreprinetti) [inline pkg/openstack/loadbalancer_test.go:642]: nit: This is not the "expected" result (which we would define in the test case), but rather the actual result. Consider renaming this variable to just "result" to avoid confusion in the matchers.

```suggestion
			result := lbaas.createLoadBalancerStatus(tt.args.service, tt.args.svcConf, tt.args.addr)
```

**Review by pierreprinetti — COMMENTED** [review]
Looks good to me! Please apply the little fix in the inline comments, then squash, and we're ready to merge. Thank you!

**Comment by sakshi-1505** [comment]
PLease review again @pierreprinetti , I have made all requested changes

**Pull request merged**
